ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/validation/ModelConformanceValidator.scala

```diff
@@ -40,99 +44,107 @@ import com.daml.logging.LoggingContext.withEnrichedLoggingContext
+          .left
+          .map(error =>
+            rejections.buildRejectionStep(
+              transactionEntry,
+              rejectionReasonForValidationError(error),
+              commitContext.recordTime,
+            )
+          )
+      } yield ()
+      stepResult.fold(identity, _ => StepContinue(transactionEntry))
+    } catch {
+      case err: Err.MissingInputState =>
+        logger.warn(
+          "Model conformance validation failed due to a missing input state (most likely due to invalid state on the participant)."
+        )
+        rejections.buildRejectionStep(
+          transactionEntry,
+          RejectionReasonV0.Disputed(err.getMessage),
+          commitContext.recordTime,
+        )
+    }
   }
 
-  // Helper to lookup contract instances. We verify the activeness of
-  // contract instances here. Since we look up every contract that was
+  // Helper to lookup contract instances. Since we look up every contract that was
   // an input to a transaction, we do not need to verify the inputs separately.
-  private def lookupContract(
-      transactionEntry: DamlTransactionEntrySummary,
-      commitContext: CommitContext,
-  )(
-      coid: Value.ContractId
-  ): Option[Value.ContractInst[Value.VersionedValue[Value.ContractId]]] = {
+  private[validation] def lookupContract(
+      commitContext: CommitContext
+  )(coid: Value.ContractId): Option[Value.ContractInst[Value.VersionedValue[Value.ContractId]]] = {
     val stateKey = contractIdToStateKey(coid)
-    for {
-      // Fetch the state of the contract so that activeness can be checked.
-      // There is the possibility that the reinterpretation of the transaction yields a different
-      // result in a LookupByKey than the original transaction. This means that the contract state data for the
-      // contractId pointed to by that contractKey might not have been preloaded into the input state map.
-      // This is not a problem because after the transaction reinterpretation, we compare the original
-      // transaction with the reinterpreted one, and the LookupByKey node will not match.
-      // Additionally, all contract keys are checked to uphold causal monotonicity.
-      contractState <- commitContext.read(stateKey).map(_.getContractState)
-      if contractIsActive(transactionEntry, contractState)
-      contract = Conversions.decodeContractInstance(contractState.getContractInstance)
-    } yield contract
+    // There is the possibility that the reinterpretation of the transaction yields a different
+    // result in a LookupByKey than the original transaction. This means that the contract state data for the
+    // contractId pointed to by that contractKey might not have been preloaded into the input state map.
+    // This is not a problem because after the transaction reinterpretation, we compare the original
+    // transaction with the reinterpreted one, and the LookupByKey node will not match.
+    commitContext
+      .read(stateKey)
```

Comment:
A question from me and @fabiotudone-da: what if a contract is not in the state anymore?

Reply:
It can only not be in the state, if the contract was archived and pruned on the committer.
The question is then, why was the participant able to produce such a transaction? For which the answer is: it used a divulged contract that appeared as active, because it didn't learn about the archival. On the other hand, using divulged contracts for interpretation has been [deprecated recently](https://docs.daml.com/support/component-statuses.html#ledger-api).

Technically I think it should result in an `Inconsistent` error though.

Reply:
Thanks! I was also thinking about a racy situation.

Reply:
I don’t think you can get it through a race:

1. The participant will only use a contract if it’s been committed.
2. If it’s been committed it’s in the committer state and it stays there until committer pruning.

Reply:
Thanks again, I get it now 👍🏻